src/server/core/game/database-manager.ts

```diff
@@ -44,6 +44,22 @@ export class DatabaseManager {
     await this.manager.updateMany(Player, { }, { $set: { loggedIn: false } });
   }
 
+  public async checkForIPBan(ip: string) {
+    return await this.manager.findOne(Player, { lastIP: ip, banned: true });
```

Comment:
instead of tracking lastIp, track all IP in an array (this query will still work, since mongo lets you query string arrays like that)

this has the bonus of tracking every IP a person uses and banning all of them